decred/tests/unit/dcr/wire/test_netaddress.py

```diff
@@ -0,0 +1,167 @@
+    na.addService(wire.SFNodeNetwork)
+    assert na.services == wire.SFNodeNetwork
+    assert na.hasService(wire.SFNodeNetwork)
+
+    # Ensure max payload is expected value for latest protocol version.
+    wantPayload = 30
+    maxPayload = netaddress.MaxNetAddressPayload
+    assert maxPayload == wantPayload
+
+    # Ensure max payload length is not more than MaxMessagePayload.
+    assert maxPayload <= wire.MaxMessagePayload
+
+
+def test_NetAddressWire():
+    """
+    test the NetAddress wire encode and decode for various protocol versions and
+    timestamp flag combinations.
+    """
+    # baseNetAddr is used in the various tests as a baseline NetAddress.
+    baseNetAddr = netaddress.NetAddress(
+        ip="127.0.0.1",
+        port=8333,
+        services=wire.SFNodeNetwork,
+        stamp=0x495FAB29,  # 2009-01-03 12:15:05 -0600 CST
+    )
+
+    # baseNetAddrNoTS is baseNetAddr with a zero value for the timestamp.
+    baseNetAddrNoTS = netaddress.NetAddress(
+        ip=baseNetAddr.ip,
+        port=baseNetAddr.port,
+        services=baseNetAddr.services,
+        stamp=0,
+    )
+
+    # fmt: off
+    # baseNetAddrEncoded is the wire encoded bytes of baseNetAddr.
+    baseNetAddrEncoded = ByteArray([
+        0x29, 0xab, 0x5f, 0x49, # Timestamp
+        0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, # SFNodeNetwork
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0xff, 0xff, 0x7f, 0x00, 0x00, 0x01, # IP 127.0.0.1
+        0x20, 0x8d, # Port 8333 in big-endian
+    ])
+
+    # baseNetAddrNoTSEncoded is the wire encoded bytes of baseNetAddrNoTS.
+    baseNetAddrNoTSEncoded = ByteArray([
+        # No timestamp
+        0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, # SFNodeNetwork
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0xff, 0xff, 0x7f, 0x00, 0x00, 0x01, # IP 127.0.0.1
+        0x20, 0x8d, # Port 8333 in big-endian
+    ])
+    # fmt: on
+
+    """
+    addrIn NetAddress to encode
+    out    Expected decoded NetAddress
+    ts     Include timestamp?
+    buf    Wire encoding
+    pver   Protoc
+    """
+
+    tests = [
+        # Latest protocol version without ts flag.
+        dict(
+            addrIn=baseNetAddr,
+            out=baseNetAddrNoTS,
+            ts=False,
+            buf=baseNetAddrNoTSEncoded,
+            pver=wire.ProtocolVersion,
+        ),
+        # Latest protocol version with ts flag.
+        dict(
+            addrIn=baseNetAddr,
+            out=baseNetAddr,
+            ts=True,
+            buf=baseNetAddrEncoded,
+            pver=wire.ProtocolVersion,
+        ),
+    ]
+
+    for test in tests:
+        # Encode to wire format.
+        b = netaddress.writeNetAddress(test["addrIn"], test["ts"])
+
+        assert b == test["buf"]
+
+        # Decode the message from wire format.
+        na = netaddress.readNetAddress(test["buf"], test["ts"])
+
+        assert byteIP4 == test["out"].ip
```

Comment:
Remove blank line, and the assertion between two constant values too.

Reply:
This assertion is testing that the string IP input is properly converted to bytes.